prototype/docs/core/metrics.md

```diff
@@ -0,0 +1,290 @@
+Note that this might unregister gauges registered by other components.
+
+```java
+package org.eclipse.sensinact.tutorials.metrics.oshi;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.eclipse.sensinact.core.metrics.IMetricsManager;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+
+import oshi.SystemInfo;
+import oshi.hardware.CentralProcessor;
+import oshi.hardware.HardwareAbstractionLayer;
+
+/**
+ * Oshi-based system metrics
+ */
+@Component
+public class OshiMetrics {
+
+    /**
+     * This component requires the core metrics service
+     */
+    @Reference
+    private IMetricsManager metrics;
+
+    /**
+     * Component is activated, the metrics service is available
+     */
+    @Activate
+    void activate() {
+        // Get access to system information via Oshi
+        final SystemInfo si = new SystemInfo();
+        final HardwareAbstractionLayer hal = si.getHardware();
+        final CentralProcessor cpu = hal.getProcessor();
+
+        // Get the initial CPU load ticks
+        final AtomicReference<long[]> previousTicks = new AtomicReference<long[]>(cpu.getSystemCpuLoadTicks());
+
+        // Register a callback method as a gauge. It will be called during each report generation
+        metrics.registerGauge("oshi.cpu", () -> {
+            // Get the previous ticks and store the new ones
+            final long[] oldTicks = previousTicks.getAndSet(cpu.getSystemCpuLoadTicks());
+
+            // Compute the system CPU load compared the previous ticks
+            return cpu.getSystemCpuLoadBetweenTicks(oldTicks);
+        });
+
+        // Register another callback method as a gauge. It will be called during each report generation
+        metrics.registerGauge("oshi.memory.available", () -> {
+            // Return the current available memory size
+            return hal.getMemory().getAvailable();
+        });
+    }
+
+    /**
+     * The component is deactivated, the metrics service is still available
+     */
+    @Deactivate
+    void deactivate() {
+        // Unregister all gauges whose name starts with "oshi."
+        metrics.unregisterGaugesByPrefix("oshi.");
+    }
+}
+```
+
+### Use timers
+
+To use sensiNact metrics timers, it is recommended to use the try-with-resources pattern.
+The timer is started as soon as it is returned by the metrics service and stopped when its `close()` method is called.
+
+Here is an example usage:
+```java
+try (IMetricTimer timer = metrics.withTimer("tutorial.metric.name")) {
+    executeTask();
+}
+```
+
+Unlike other statistics, you cannot reuse a timer instance.
+
+### Use other metrics
+
+Counters and histograms can be used inline.
+It is not recommended to keep it as a class member as the returned instance depends on the metrics activation.
```

Comment:
I think that this should be fixed in the implementation of the metrics, otherwise it just makes the usage harder.